orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartAwsCodeBuildTask.java

```diff
@@ -51,4 +72,82 @@ public TaskResult execute(@Nonnull Stage stage) {
     context.put("buildInfo", execution);
     return TaskResult.builder(ExecutionStatus.SUCCEEDED).context(context).build();
   }
+
+  private void appendProjectName(Map<String, Object> requestInput, String projectName) {
+    requestInput.put(PROJECT_NAME, projectName);
+  }
+
+  private void appendSource(
+      Map<String, Object> requestInput, Artifact artifact, String sourceType) {
+    if (sourceType != null && !sourceType.equals("")) {
+      requestInput.put(SOURCE_TYPE, sourceType);
+    }
+    switch (artifact.getType()) {
+      case "s3/object":
+        requestInput.putIfAbsent(SOURCE_TYPE, "S3");
+        String s3Location = artifact.getReference();
+        requestInput.put(
+            SOURCE_LOCATION, s3Location.startsWith("s3://") ? s3Location.substring(5) : s3Location);
+        break;
+      case "git/repo":
+        requestInput.put(SOURCE_LOCATION, artifact.getReference());
+        if (artifact.getReference().matches("^(http(s)?://)github.com/(.*)$")) {
+          requestInput.putIfAbsent(SOURCE_TYPE, "GITHUB");
+        } else if (artifact.getReference().matches("^(http(s)?://)(.*@)?bitbucket.org/(.*)$")) {
+          requestInput.putIfAbsent(SOURCE_TYPE, "BITBUCKET");
+        } else if (artifact
+            .getReference()
+            .matches("^https://git-codecommit.(.*).amazonaws.com/(.*)$")) {
+          requestInput.putIfAbsent(SOURCE_TYPE, "CODECOMMIT");
+        } else {
+          throw new IllegalStateException("Source type could not be inferred from location");
+        }
```

Comment:
This will still throw, even if I provided sourceType explicitly.  This should check if source type is already set (and unit test should test this)

Reply:
Fixed